### PR TITLE
[SLDEV-18523] Add tokenFile CLI parameter

### PR DESCRIPTION
**Pull request by ursulaflorianczyk**

### CODE REVIEW

File: src/main/kotlin/io/sealights/tool/source/SourceCodeLineReader.kt [hunk 1]
```diff
@@ -27,15 +29,24 @@ class SourceCodeLineReader {
...
     private fun readFileLines(physicalPath: FileName): List<String> {
-        val file = File(physicalPath)
         val content = mutableListOf<String>()
-
-        if (file.exists()) {
-            file.forEachLine { line ->
-                content.add(line)
+        try {
+            val path = Paths.get(Configuration.workspace, physicalPath).normalize()
+            val file = path.toFile()
```
Comment (gregoryfuta) [inline src/main/kotlin/io/sealights/tool/source/SourceCodeLineReader.kt:35]: you can just use `path.toFile().readLines()` (simplifying example)

**Review by gregoryfuta — COMMENTED** [review]
Let's talk about things related to `Configuration.kt` tomorrow.

File: src/main/kotlin/io/sealights/tool/source/SourceCodeLineReader.kt (hunk 1 above)
Comment (ursulaflorianczyk) [reply src/main/kotlin/io/sealights/tool/source/SourceCodeLineReader.kt:35]: I want to check if file exists

File: src/main/kotlin/io/sealights/tool/configuration/ApplicationArgParser.kt [hunk 2]
```diff
@@ -30,13 +36,23 @@ class ApplicationArgParser(argParser: ArgParser) {
...
+        help = "componentName - required if the integrationBuildSessionId is set"
     ).default("")
 
-//    fun set(token: String, startCommit: String, endCommit: String = "HEAD", workspace: String = ".") {
-//        Configuration.token = token
-//        Configuration.workspace = workspace
-//        Configuration.startCommit = startCommit
-//        Configuration.endCommit = endCommit
-//    }
+    fun getTokenVale(): String?{
```
[the remark this answers is not in the document]
Comment (ursulaflorianczyk) [reply src/main/kotlin/io/sealights/tool/configuration/ApplicationArgParser.kt:42]: refactored according your suggestion

**Review by gregoryfuta — APPROVED** [review]

File: src/main/kotlin/io/sealights/tool/Application.kt [hunk 3]
```diff
@@ -22,21 +21,15 @@ import io.sealights.tool.source.SourceCodeLineReader
...
     println("Sealights Line Level Coverage details")
 
-    val tokenResolver = TokenResolver()
-
-    ArgParser(args).parseInto(::ApplicationArgParser)
-        .run {
-            Configuration.build(this, tokenResolver)
-            Configuration.printConfiguration()
-            Configuration.validate()
+    Configuration.buildAndValidate(args, TokenResolver())
```
[the remark this answers is not in the document]
Comment (gregoryfuta) [reply src/main/kotlin/io/sealights/tool/Application.kt:24]: Let's do that later.

**Pull request merged**